ivy_models/clip/misc.py

```diff
@@ -0,0 +1,193 @@
+import warnings
+from typing import Union, List
+from pkg_resources import packaging
+
+
+import ivy
+import torch
+from torchvision.transforms import Compose, Resize, CenterCrop, ToTensor, Normalize
+from PIL import Image
+
+from .simple_tokenizer import SimpleTokenizer as _Tokenizer
+
+try:
+    from torchvision.transforms import InterpolationMode
+
+    BICUBIC = InterpolationMode.BICUBIC
+except ImportError:
+    BICUBIC = Image.BICUBIC
+
+
+if packaging.version.parse(torch.__version__) < packaging.version.parse("1.7.1"):
+    warnings.warn("PyTorch version 1.7.1 or higher is recommended")
+
+
+_tokenizer = _Tokenizer()
+
+_MODELS = {
+    "RN50": "https://openaipublic.azureedge.net/clip/models/afeb0e10f9e5a86da6080e35cf09123aca3b358a0c3e3b6c78a7b63bc04b6762/RN50.pt",
+    "RN101": "https://openaipublic.azureedge.net/clip/models/8fa8567bab74a42d41c5915025a8e4538c3bdbe8804a470a72f30b0d94fab599/RN101.pt",
+    "RN50x4": "https://openaipublic.azureedge.net/clip/models/7e526bd135e493cef0776de27d5f42653e6b4c8bf9e0f653bb11773263205fdd/RN50x4.pt",
+    "RN50x16": "https://openaipublic.azureedge.net/clip/models/52378b407f34354e150460fe41077663dd5b39c54cd0bfd2b27167a4a06ec9aa/RN50x16.pt",
+    "RN50x64": "https://openaipublic.azureedge.net/clip/models/be1cfb55d75a9666199fb2206c106743da0f6468c9d327f3e0d0a543a9919d9c/RN50x64.pt",
+    "ViT-B/32": "https://openaipublic.azureedge.net/clip/models/40d365715913c9da98579312b702a82c18be219cc2a73407c4526f58eba950af/ViT-B-32.pt",
+    "ViT-B/16": "https://openaipublic.azureedge.net/clip/models/5806e77cd80f8b59890b7e101eabd078d9fb84e6937f9e85e4ecb61988df416f/ViT-B-16.pt",
+    "ViT-L/14": "https://openaipublic.azureedge.net/clip/models/b8cca3fd41ae0c99ba7e8951adf17d267cdb84cd88be6f7c2e0eca1737a03836/ViT-L-14.pt",
+    "ViT-L/14@336px": "https://openaipublic.azureedge.net/clip/models/3035c92b350959924f9f00213499208652fc7ea050643e8b385c2dac08641f02/ViT-L-14-336px.pt",
+}
+
+
+def _convert_image_to_rgb(image):
+    return image.convert("RGB")
+
+
+def _transform(n_px):
```

Comment:
there's an equivalent in `ivy_models_tests/image_helpers.py`
can we also refactor some of the helpers here to that file or folder? thanks!
ps: the image helpers were initially used during tests only, but if it is deemed better in `ivy_models`, please feel free to refactor this in another PR

Reply:
The CLIP image transforms depend on the variant of the model that we load, so I thought it was better to handle it there for convenience. Also, as you mentioned, I think it'd be better to have some helpers to load the appropriate image preprocessing pipeline in `ivy_models` so that users can easily load them directly. Currently, users will have to write the appropriate image processing pipeline for each model that they load and that won't be cool for the UX.

Reply:
we have the transform pipeline there, this function looks quite similar to that imo, the resize and centercrop args are configurable as well. do you think it can be improved with the rgb flag? Thanks!
ref: https://github.com/unifyai/models/blob/c87584bf7be32c8b2ed58e3978452645e9ff36a0/ivy_models_tests/helpers/image_helpers.py#L30